docs/multiprocessing.rst

```diff
@@ -0,0 +1,7 @@
+.. note::
```

Comment:
@plamut Is this note appropriate for BigQuery? If not I'll add this RST to the list of synth excludes

Reply:
BigQuery client uses both the BQ Storage Client and the `google.cloud.Client` (it inherits from it). Do you know if the latter is thread-safe?
If nothing else, the note _"this client uses grpcio"_ can be misleading, as it can appear that the client uses _only_ `grpcio` for the transport.

The BigQuery client itself does not seem to have much state, though, most of the stuff is just initialized in `__init__()`. It does have a [close()](https://github.com/googleapis/python-bigquery/blob/3869e34f3eff6fabca21c397e3f1cbc368ec880e/google/cloud/bigquery/client.py#L201-L211) method that closes the underlying transport, but deciding when to close the shared client instance is the responsibility of the developer IMO.

@tswast Do you maybe remember if there have been any thread safety issues in the past and if the BigQuery client was actually designed for thread safety?

Reply:
You're right that the client doesn't have much state other than a few "default" config values and the connection info. The only mutable state I can think of is the Credentials object (fetches new access token every time one expires). The `Job` objects _do_ have mutable state for tracking if the job is finished or not, but I wouldn't expect those to be used across threads.

In the long-long ago it definitely wasn't thread safe due to use of httplib2 (https://github.com/googleapis/google-cloud-python/issues/1346), but now we use `requests`, so those issues should have been resolved.

Reply:
It looks like there is already a bigquery specific note related to multiprocessing (I should have checked before I asked, sorry! 😅 ).  https://github.com/googleapis/python-bigquery/blob/master/docs/index.rst

Updated synth.py to not copy the generic version.